Server crashes because module named 'yaml' not found

### Summary

The `ttnn-visualizer` crashes because of missing module after clean install.

To reproduce, start by installing a clean version of the application:

```sh
# Create fresh venv and start it
python3.12 -m venv .
source bin/activate

# Install latest wheel
python3.12 -m pip install ttnn_visualizer-0.17.1-py3-none-any.whl
```

Starting the application will fail with the following error:

```sh
(ttnn-visualizer) % ttnn-visualizer
ENVIRONMENT:
APPLICATION_DIR=/Users/esmal/Code/ttnn-visualizer/lib/python3.12/site-packages
DEBUG=False
DEV_SERVER_HOST=localhost
DEV_SERVER_PORT=5173
GUNICORN_APP_MODULE=ttnn_visualizer.app:create_app()
GUNICORN_BIND=0.0.0.0:8000
GUNICORN_WORKERS=1
GUNICORN_WORKER_CLASS=gevent
HOST=0.0.0.0
LAUNCH_BROWSER_ON_START=True
LOCAL_DATA_DIRECTORY=/Users/esmal/Code/ttnn-visualizer/lib/python3.12/site-packages/ttnn_visualizer/data/local
PORT=8000
PRINT_ENV=True
REMOTE_DATA_DIRECTORY=/Users/esmal/Code/ttnn-visualizer/lib/python3.12/site-packages/ttnn_visualizer/data/remote
REPORT_DATA_DIRECTORY=/Users/esmal/Code/ttnn-visualizer/lib/python3.12/site-packages/ttnn_visualizer/data
SECRET_KEY=90909
SEND_FILE_MAX_AGE_DEFAULT=0
SESSION_COOKIE_SAMESITE=Lax
SESSION_COOKIE_SECURE=False
SQLALCHEMY_DATABASE_URI=sqlite:////Users/esmal/Code/ttnn-visualizer/lib/python3.12/site-packages/ttnn_0.14.2.db
SQLALCHEMY_ENGINE_OPTIONS={'pool_size': 10, 'max_overflow': 20, 'pool_timeout': 30}
SQLALCHEMY_TRACK_MODIFICATIONS=False
SQLITE_DB_PATH=db.sqlite
STATIC_ASSETS_DIR=/Users/esmal/Code/ttnn-visualizer/lib/python3.12/site-packages/ttnn_visualizer/static
TESTING=False
TEST_CONFIG_FILE=config.json
USE_WEBSOCKETS=True
VERSION=0.14.2
[2025-01-20 12:51:08 -0500] [55867] [INFO] Starting gunicorn 22.0.0
[2025-01-20 12:51:08 -0500] [55867] [INFO] Listening at: http://0.0.0.0:8000 (55867)
[2025-01-20 12:51:08 -0500] [55867] [INFO] Using worker: gevent
[2025-01-20 12:51:08 -0500] [55868] [INFO] Booting worker with pid: 55868
[2025-01-20 12:51:08 -0500] [55868] [ERROR] Exception in worker process
Traceback (most recent call last):
  File "/Users/esmal/Code/ttnn-visualizer/lib/python3.12/site-packages/gunicorn/arbiter.py", line 609, in spawn_worker
    worker.init_process()
  File "/Users/esmal/Code/ttnn-visualizer/lib/python3.12/site-packages/gunicorn/workers/ggevent.py", line 147, in init_process
    super().init_process()
  File "/Users/esmal/Code/ttnn-visualizer/lib/python3.12/site-packages/gunicorn/workers/base.py", line 134, in init_process
    self.load_wsgi()
  File "/Users/esmal/Code/ttnn-visualizer/lib/python3.12/site-packages/gunicorn/workers/base.py", line 146, in load_wsgi
    self.wsgi = self.app.wsgi()
                ^^^^^^^^^^^^^^^
  File "/Users/esmal/Code/ttnn-visualizer/lib/python3.12/site-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
                    ^^^^^^^^^^^
  File "/Users/esmal/Code/ttnn-visualizer/lib/python3.12/site-packages/gunicorn/app/wsgiapp.py", line 58, in load
    return self.load_wsgiapp()
           ^^^^^^^^^^^^^^^^^^^
  File "/Users/esmal/Code/ttnn-visualizer/lib/python3.12/site-packages/gunicorn/app/wsgiapp.py", line 48, in load_wsgiapp
    return util.import_app(self.app_uri)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/esmal/Code/ttnn-visualizer/lib/python3.12/site-packages/gunicorn/util.py", line 424, in import_app
    app = app(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^
  File "/Users/esmal/Code/ttnn-visualizer/lib/python3.12/site-packages/ttnn_visualizer/app.py", line 29, in create_app
    from ttnn_visualizer.views import api
  File "/Users/esmal/Code/ttnn-visualizer/lib/python3.12/site-packages/ttnn_visualizer/views.py", line 13, in <module>
    import yaml
ModuleNotFoundError: No module named 'yaml'
[2025-01-20 12:51:08 -0500] [55868] [INFO] Worker exiting (pid: 55868)
[2025-01-20 12:51:08 -0500] [55867] [ERROR] Worker (pid:55868) exited with code 3
[2025-01-20 12:51:08 -0500] [55867] [ERROR] Shutting down: Master
[2025-01-20 12:51:08 -0500] [55867] [ERROR] Reason: Worker failed to boot.
```

Manually installing `pyyaml` works as a temporary workaround.

Here is the information about my platform:

```
Darwin Evan-Smal's-Mac 24.1.0 Darwin Kernel Version 24.1.0: Thu Oct 10 21:03:11 PDT 2024
Python 3.12.8
rustc 1.84.0 (9fc6b4312 2025-01-07)
```

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -19,7 +19,8 @@ dependencies = [
     "python-dotenv==1.0.1",
     "sqlalchemy==2.0.34",
     "flask-socketio==5.4.1",
-    "flask-sqlalchemy==3.1.1"
+    "flask-sqlalchemy==3.1.1",
+    "PyYAML==6.0.2"
 ]
 
 classifiers = [

